elastic/kibana
# [ResponseOps][Alerts] Fix stack alerts plugin API and page permissions

## Summary

- ~~Adds the `rac` api privilege to `stackAlerts`  plugin's Kibana feature registration~~ (done in https://github.com/elastic/kibana/pull/193948)
- Extends the `fetchAlertsIndexNames` API to return a boolean flag that indicates if the end user (not Kibana user) has (at least) read index privileges on the alert index name returned by the API, regardless of the alerting authorization
- Uses the flag mentioned in the previous point to only render the alert filter controls bar if the user has personal read access to the indices targeted by the current feature ids
- Only uses feature ids the user has access to for the stack alerts page (shouldn't be necessary, but it restricts the resulting index pattern, which is helpful to better identify when to show/hide the filter controls)
- Fixes a small issue that caused `useAlertsDataView` to return an intermediate empty data view object while loading

## To verify

1. Create rules that fire alerts in Stack management, Observability and Security
2. Wait for alerts to be created
3. Check that the stack alerts page renders correctly and shows alerts coherent with the feature filter and other KQL bar filters
4. Create a role with only `Stack Management > Rules : Read` privilege
5. Create a user with that role
6. In another window, open Kibana with the newly created user
7. Check that the Stack Management > Alerts page renders correctly, only showing alerts from `stackAlerts` rule types and without the filter controls bar
8. Add read privilege to the `.alerts-*` index to the role
9. Reload the alerts page as unprivileged user and check that the filter controls are now visible
10. Gradually add other solution permissions and check that the alerts shown in the page change accordingly

### References

Fixes #186451

### Checklist

Delete any items that are not applicable to this PR.
